presto-main/src/main/java/com/facebook/presto/operator/aggregation/MaxAggregation.java

```diff
@@ -0,0 +1,397 @@
+
+    @Override
+    public Signature getSignature()
+    {
+        return SIGNATURE;
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Returns the maximum value of the argument";
+    }
+
+    @Override
+    public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type type = types.get("E");
+        MethodHandle greaterThanMethodHandle = functionRegistry.resolveOperator(OperatorType.GREATER_THAN, ImmutableList.of(type, type)).getMethodHandle();
+        Signature signature = new Signature(NAME, type.getTypeSignature(), type.getTypeSignature());
+        InternalAggregationFunction aggregation = generateAggregation(type, greaterThanMethodHandle);
+        return new FunctionInfo(signature, getDescription(), aggregation.getIntermediateType().getTypeSignature(), aggregation, false);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type type, MethodHandle greaterThanMethodHandle)
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(MaxAggregation.class.getClassLoader());
+
+        List<Type> inputTypes = ImmutableList.of(type);
+
+        AccumulatorStateSerializer stateSerializer;
+        AccumulatorStateFactory stateFactory;
+        Method inputFunction;
+        Method outputFunction;
+        Method combineFunction;
+
+        if (type.getJavaType() == long.class) {
+            stateFactory = new FactoryForLong(type, greaterThanMethodHandle);
+            stateSerializer = new StateSerializerForLong();
+            inputFunction = LONG_INPUT_FUNCTION;
+            outputFunction = LONG_OUTPUT_FUNCTION;
+            combineFunction = LONG_COMBINE_FUNCTION;
+        }
+        else {
+            throw new UnsupportedOperationException("Not yet implemented");
+        }
+
+        Type intermediateType = stateSerializer.getSerializedType();
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, type, inputTypes),
+                createInputParameterMetadata(type),
+                inputFunction,
+                null,
+                null,
+                combineFunction,
+                outputFunction,
+                LongState.class,
+                stateSerializer,
+                stateFactory,
+                type,
+                false);
+
+        GenericAccumulatorFactoryBinder factory = new AccumulatorCompiler().generateAccumulatorFactoryBinder(metadata, classLoader);
+        return new InternalAggregationFunction(NAME, inputTypes, intermediateType, type, true, false, factory);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type type)
+    {
+        return ImmutableList.of(
+                new ParameterMetadata(STATE),
+                new ParameterMetadata(INPUT_CHANNEL, type));
+    }
+
+    private static interface LongState
```

Comment:
Can't we use `com.facebook.presto.operator.aggregation.state.LongState` for this?